queries.sql

```diff
@@ -14,4 +14,41 @@ SELECT * FROM animals WHERE neutered;
 
 SELECT * FROM animals WHERE NOT name='Gabumon';
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+DELETE animals WHERE date_of_birth > '2022/1/1';
+SAVEPOINT DLT_YNGST;
+UPDATE animals SET weight_kg =weight_kg * -1;
+ROLLBACK TO SAVEPOINT DLT_YNGST;
+UPDATE animals SET weight_kg =weight_kg * -1 WHERE SIGN(weight_kg)=-1;
```

Comment:
- [ ] It would be better if our query answers the following question:

1. UPDATE all animals' weights that are negative to be their weight multiplied by -1. 
In our case, we are only multiplying weights that are only `-1`, how about `-2, -3, -4, -5 ...`?
Instead we can achieve that by making the condition as `WHERE  SIGN(weight_kg) < 0. Anything less than zero is a negative number